### PR TITLE
uim: disable qt support

### PR DESCRIPTION
**Pull request by Atemu**
https://github.com/NixOS/nixpkgs/pull/174634 removed qt4 but enabled the (previosly disabled) qt5, breaking the build.

Disable QT5 again by disabling qt all together.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
cc @ericsagnes (do you still maintain this package?)

**Comment by ajs124** [comment]
can we maybe fix the qt5 support by adding `wrapQtAppsHook`?

File: pkgs/tools/inputmethods/uim/default.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@
 , withGtk ? true
 , withGtk2 ? withGtk, gtk2 ? null
 , withGtk3 ? withGtk, gtk3 ? null
-, withQt ? true
+, withQt ? false # qt5 is broken and qt4 removed -> No qt support possible
```
Comment (doronbehar) [inline pkgs/tools/inputmethods/uim/default.nix:9]: Seriously? `qt5.qtbase` is broken?

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/tools/inputmethods/uim/default.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/tools/inputmethods/uim/default.nix:9]: qt5 support in uim is broken in Nixpkgs.

Comment (doronbehar) [reply pkgs/tools/inputmethods/uim/default.nix:9]: TBH the comment you phrased seems more confusing then explaining. First of all, there is no reference to qt4 in the expression. The `withQt` seems to me like an old argument that was replaced / superseded at some point by `withQt5`. Hence, I'd remove completely `withQt` in favor of `withQt5`. Hence I'd rephrase the comment as follows:

```
# qt5 support is broken in uim
```

No need to mention qt4, and the word support is important here.

**Comment by doronbehar** [comment]
> can we maybe fix the qt5 support by adding `wrapQtAppsHook`?

@Atemu could it be that this is why some users experience this qt5 support as broken?

**Comment by Atemu** [comment]
I'll clarify the comment but I won't pursue actually fixing the package.  
I have no idea how it works or how I'd even test it works.

All I know is that it transitively breaks mlterm and why.

File: pkgs/tools/inputmethods/uim/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/tools/inputmethods/uim/default.nix:9]: qt5 support is not broken if it's just missing the wrapper from nativeBuildInputs because it can be fixed by adding it.

**Comment by doronbehar** [comment]
I too have no idea what is the purpose of this package. Secondly, before https://github.com/NixOS/nixpkgs/commit/308baf304984fb5bb94f0b69f4b4dc8d5030405d , qt5 support was never enabled, and for sure it did not work since forever. However It is not that hard to fix it, and I would also say that its not accurate to say that `qt5 support is broken in uim`. Hence I propose this change:

```diff
diff --git i/pkgs/tools/inputmethods/uim/default.nix w/pkgs/tools/inputmethods/uim/default.nix
index 70d3e8e8725c..2ad8e60e8e92 100644
--- i/pkgs/tools/inputmethods/uim/default.nix
+++ w/pkgs/tools/inputmethods/uim/default.nix
@@ -44,6 +44,8 @@ stdenv.mkDerivation rec {
 
     ruby # used by sigscheme build to generate function tables
     librsvg # used by uim build to generate png pixmaps from svg
+  ] ++ lib.optionals withQt5 [
+    qt5.wrapQtAppsHook
   ];
 
   buildInputs = [
@@ -52,7 +54,7 @@ stdenv.mkDerivation rec {
   ++ lib.optional withAnthy anthy
   ++ lib.optional withGtk2 gtk2
   ++ lib.optional withGtk3 gtk3
-  ++ lib.optionals withQt5 [ qt5.qtbase.bin qt5.qtbase.dev ]
+  ++ lib.optionals withQt5 [ qt5.qtbase qt5.qtx11extras ]
   ++ lib.optional withLibnotify libnotify
   ++ lib.optional withSqlite sqlite
   ++ lib.optionals withNetworking [
```

We can apply the above while still disabling qt5 support by default, while removing the comment near it - because it is not correct, and the truth is that for no particular reason we disable qt5 support by default.

**Comment by Atemu** [comment]
@doronbehar feel free to open a PR on that. As I mentioned, I'm not taking ownership of qt support in uim, I'm just restoring it to the status quo.

**Comment by doronbehar** [comment]
It's just that our approaches a bit conflicting, and my solution will also return things to their status quo.. I hope you won't mind, but I pushed here my change in addition to yours, along with a change to the comment.

**Comment by Atemu** [comment]
I'm not a fan of adding dead code that I don't know what it does. Could you open your own PR for that?

**Comment by doronbehar** [comment]
> I'm not a fan of adding dead code that I don't know what it does. Could you open your own PR for that?

I can just merge this one :)

**Comment by doronbehar** [comment]
I did test that the dead code compiles successfully, observed that binaries with `-qt5` are added to `result/bin`, so I think it is good enough for the moment.

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 255091` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
    <li>uim</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
For the record^.
